GitCommands/CommitTemplateManager.cs

```diff
@@ -63,7 +76,16 @@ public CommitTemplateManager(IGitModule module)
         /// <summary>
         /// Gets the collection of all currently registered commit templates provided by plugins.
         /// </summary>
-        public IEnumerable<CommitTemplateItem> RegisteredTemplates => RegisteredTemplatesDic.Select(item => new CommitTemplateItem(item.Key, item.Value()));
+        public IEnumerable<CommitTemplateItem> RegisteredTemplates
+        {
+            get
+            {
+                lock (RegisteredTemplatesStorage)
+                {
+                    return RegisteredTemplatesStorage.Select(item => new CommitTemplateItem(item.Name, item.Text()));
```

Comment:
This is not thread safe. The returned enumerable will be evaluated outside the lock, and can still result in a "collection modified" exception.

The easiest fix is to call `ToList()` within the loop. In such a case I'd also return `IReadOnlyList<CTI>`.